core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -657,6 +681,28 @@ class BlockManagerMasterEndpoint(
     }
   }
 
+  private def getShufflePushMergerLocations(
+      numMergersNeeded: Int,
+      hostsToFilter: Set[String]): Seq[BlockManagerId] = {
+    val activeBlockManagers = blockManagerIdByExecutor.groupBy(_._2.host)
+      .mapValues(_.head).values.map(_._2).toSet
+    val filteredActiveBlockManagers = activeBlockManagers
+      .filterNot(x => hostsToFilter.contains(x.host))
+    val filteredActiveMergers = filteredActiveBlockManagers.map(
+      x => BlockManagerId(x.executorId, x.host, StorageUtils.externalShuffleServicePort(conf)))
+
+    // Enough mergers are available as part of active executors list
+    if (filteredActiveMergers.size >= numMergersNeeded) {
+      filteredActiveMergers.toSeq
+    } else {
+      // Delta mergers added from inactive mergers list to the active mergers list
+      val filteredDeadMergers = shuffleMergerLocations.values
```

Comment:
"dead mergers" / "active" / "inactive" sound confused. "dead" or "inactive" doesn't mean the merger is in bad status that can't work normally but just mean there're no executors on the same host, right?

How about renaming `filteredActiveMergers` to `filteredMergersWithExecutors` and `filteredDeadMergers` to `filteredMergersWithoutExecutors`?